bftengine/src/preprocessor/messages/PreProcessedResultMsg.cpp

```diff
@@ -0,0 +1,106 @@
+                                             const char* messageSignature,
+                                             uint32_t messageSignatureLen,
+                                             const char* resultSignaturesBuf,
+                                             const uint32_t resultSignaturesBufLen)
+    : ClientRequestMsg(sender,
+                       bftEngine::HAS_PRE_PROCESSED_FLAG,
+                       reqSeqNum,
+                       resultLength,  // check the comment in the header to
+                       result,        // understand why result is passed as request here
+                       reqTimeoutMilli,
+                       cid,
+                       spanContext,
+                       messageSignature,
+                       messageSignatureLen,
+                       resultSignaturesBufLen) {
+  msgBody_->msgType = MsgCode::PreProcessedResult;
+  // ClientRequestMsg allocates additional memory for the signatures
+  // Get pointer to it here and assert that the buffer is big enough
+  auto [pos, max_len] = getExtraBufPtr();
+  ConcordAssert(max_len >= resultSignaturesBufLen);
+
+  memcpy(pos, resultSignaturesBuf, resultSignaturesBufLen);
+}
+
+PreProcessedResultMsg::PreProcessedResultMsg(bftEngine::ClientRequestMsgHeader* body) : ClientRequestMsg(body) {}
+
+std::pair<char*, uint32_t> PreProcessedResultMsg::getResultSignaturesBuf() { return getExtraBufPtr(); }
+
+std::string PreProcessResultSignature::serializeResultSignatureList(
+    const std::list<PreProcessResultSignature>& signatures) {
+  size_t buf_len = 0;
+  for (const auto& s : signatures) {
+    buf_len += sizeof(s.sender_replica) + sizeof(uint32_t) + s.signature.size();
+  }
+
+  std::string output;
+  output.reserve(buf_len);
+
+  for (const auto& s : signatures) {
+    output.append(concordUtils::toBigEndianStringBuffer(s.sender_replica));
+    output.append(concordUtils::toBigEndianStringBuffer<uint32_t>(s.signature.size()));
+    output.append(s.signature.begin(), s.signature.end());
+  }
+
+  return output;
+}
+
+std::list<PreProcessResultSignature> PreProcessResultSignature::deserializeResultSignatureList(const char* buf,
+                                                                                               size_t len) {
+  size_t pos = 0;
+  std::list<PreProcessResultSignature> ret;
+  while (1) {
+    bftEngine::impl::NodeIdType sender_id;
+    uint32_t signature_size;
+
+    if (sizeof(sender_id) + sizeof(signature_size) > len - pos) {
+      throw std::runtime_error(
+          "PreProcessResultSignature deserialisation error - remaining buffer length less than fixed size values size");
+    }
+
+    // Read fixed size values
+    sender_id = concordUtils::fromBigEndianBuffer<bftEngine::impl::NodeIdType>(buf + pos);
+    pos += sizeof(bftEngine::impl::NodeIdType);
+    signature_size = concordUtils::fromBigEndianBuffer<uint32_t>(buf + pos);
+    pos += sizeof(uint32_t);
+
+    if (signature_size > len - pos) {
```

Comment:
This should be verified in validate() function, which looks not implemented.

Reply:
This check is "deferred" due to the VC trigger we talked about. This is the only reason not to put it in validate().

Is this acceptable after the validation code is removed from ReplicaImp?

Reply:
Yes, thanks.